TTSegmentedControl/TTSwiftUISegmentedControl.swift

```diff
@@ -0,0 +1,143 @@
+
+public struct TTSwiftUISegmentedControl: UIViewRepresentable {
+    @Binding private var selectedIndex: Int
+    private let bounceAnimationOptions: TTSegmentedControlBounceOptions?
+    private let selectionViewShadow: TTSegmentedControlShadow?
+    private let titleDistribution: TTSegmentedControl.TitleDistribution
+    private let isDragEnabled: Bool
+    private let animationOptions: TTSegmentedControlAnimationOption?
+    private let isSizeAdjustEnabled: Bool
+    private let containerBackgroundColor: UIColor
+    private let containerGradient: TTSegmentedControlGradient?
+    private let selectionViewColor: UIColor
+    private let selectionViewGradient: TTSegmentedControlGradient?
+    private let padding: CGSize
+    private let cornerRadius: CGFloat
+    private let titles: [TTSegmentedControlTitle]
+    private let isSwitchBehaviorEnabled: Bool
+    private let didBeginTouch: (() -> Void)?
+    private let didDragOverItemAtIndex: ((Int) -> Void)?
+    private let didEndTouchAtIndex: ((Int) -> Void)?
+    
+    public init(
+        titles: [TTSegmentedControlTitle] = [],
+        titleDistribution: TTSegmentedControl.TitleDistribution = .equalSpacing,
+        selectedIndex: Binding<Int> = .constant(0),
+        padding: CGSize = .init(width: 2, height: 2),
+        isDragEnabled: Bool = true,
+        animationOptions: TTSegmentedControlAnimationOption? = .init(),
+        isSizeAdjustEnabled: Bool = false,
+        containerBackgroundColor: UIColor = .white,
+        containerGradient: TTSegmentedControlGradient? = nil,
+        selectionViewColor: UIColor = .blue,
+        selectionViewGradient: TTSegmentedControlGradient? = nil,
+        selectionViewShadow: TTSegmentedControlShadow? = nil,
+        bounceAnimationOptions: TTSegmentedControlBounceOptions? = nil,
+        cornerRadius: CGFloat = -1,
+        isSwitchBehaviorEnabled: Bool = true,
+        didBeginTouch: (() -> Void)? = nil,
+        didDragOverItemAtIndex: ((Int) -> Void)? = nil,
+        didEndTouchAtIndex: ((Int) -> Void)? = nil
+    ) {
+        self.titles = titles
+        self.titleDistribution = titleDistribution
+        self._selectedIndex = selectedIndex
+        self.padding = padding
+        self.bounceAnimationOptions = bounceAnimationOptions
+        self.selectionViewShadow = selectionViewShadow
+        self.isDragEnabled = isDragEnabled
+        self.animationOptions = animationOptions
+        self.isSizeAdjustEnabled = isSizeAdjustEnabled
+        self.containerBackgroundColor = containerBackgroundColor
+        self.containerGradient = containerGradient
+        self.selectionViewColor = selectionViewColor
+        self.selectionViewGradient = selectionViewGradient
+        self.cornerRadius = cornerRadius
+        self.isSwitchBehaviorEnabled = isSwitchBehaviorEnabled
+        self.didBeginTouch = didBeginTouch
+        self.didDragOverItemAtIndex = didDragOverItemAtIndex
+        self.didEndTouchAtIndex = didEndTouchAtIndex
+    }
```

Comment:
@dumitruigor  Oare nu ar fi o idee mai buna pentru interfata de SwiftUI sa facem toti paramaterii ca modifiers?

Reply:
@EfraimB Ca de exemplu?

Reply:
Nu doar pentru cazurile astea 3. Ci pentru fiecare propietate in parte.